File: fixtures/scale/dart/large/lib/ciphers/adfgvx/adfgvx_suite.dart
import '../../core/cipher.dart';
import '../../core/cipher_suite.dart';
import '../../core/suite_result.dart';
import '../../verify/round_trip.dart';
import 'adfgvx_cipher.dart';
import 'adfgvx_key.dart';
import 'adfgvx_vectors.dart';

class AdfgvxSuite implements CipherSuite {
  @override
  String name() => 'adfgvx';

  @override
  String category() => 'cipher';

  @override
  SuiteResult run() {
    final Cipher cipher = AdfgvxCipher(AdfgvxKey.defaultKey());
    var passed = 0;
    var failed = 0;
    final start = DateTime.now();
    for (final sample in adfgvxSamples()) {
      if (roundTripCheck(cipher, sample)) {
        passed++;
      } else {
        failed++;
      }
    }
    final elapsed = DateTime.now().difference(start).inMicroseconds;
    return SuiteResult(name(), passed, failed, elapsed);
  }
}

File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/adfgvx/AdfgvxSuite.kt
package cryptobench.ciphers.adfgvx

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.roundTrip

class AdfgvxSuite : CipherSuite {
    override fun name(): String = "adfgvx"

    override fun run(): SuiteResult {
        val cipher: Cipher = AdfgvxCipher(AdfgvxKey.default())
        var passed = 0
        var failed = 0
        val start = System.nanoTime()
        for (sample in AdfgvxVectors.samples()) {
            if (roundTrip(cipher, sample)) passed++ else failed++
        }
        return SuiteResult(name(), passed, failed, System.nanoTime() - start)
    }
}
